src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -379,6 +384,169 @@ public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationT
             return keysMigrated.Count;
         }
 
+        /// <summary>
+        /// Gets Value for given Key from RC state provider
+        /// </summary>
+        /// <param name="key">Key to fetch</param>
+        /// <param name="cancellationToken">cancellationToken</param>
+        /// <returns>Value for given key</returns>
+        public async Task<byte[]> GetValueByKeyAsync(string key, CancellationToken cancellationToken)
+        {
+            using (var tx = this.rcStateProvider.GetStateManager().CreateTransaction())
+            {
+                IReliableDictionary2<string, byte[]> dictionary = null;
+                if (key.StartsWith("@@"))
+                {
+                    dictionary = this.rcStateProvider.GetActorPresenceDictionary();
+                }
+                else if (key.StartsWith("Actor"))
+                {
+                    var startIndex = this.GetNthIndex(key, '_', 2);
+                    var endIndex = this.GetNthIndex(key, '_', 3);
+                    var actorId = new ActorId(key.Substring(startIndex + 1, endIndex - startIndex - 1));
+                    dictionary = this.rcStateProvider.GetActorStateDictionary(actorId);
+                }
+                else if (key.StartsWith("RC@@"))
+                {
+                    dictionary = this.rcStateProvider.GetReminderCompletedDictionary();
+                }
+                else if (key.Equals("Timestamp_VLTM"))
+                {
+                    dictionary = this.rcStateProvider.GetLogicalTimeDictionary();
+                }
+                else if (key.StartsWith("Reminder"))
+                {
+                    var startIndex = this.GetNthIndex(key, '_', 2);
+                    var endIndex = this.GetNthIndex(key, '_', 3);
+                    var actorId = new ActorId(key.Substring(startIndex + 1, endIndex - startIndex - 1));
+                    dictionary = this.rcStateProvider.GetReminderDictionary(actorId);
+                }
+                else
+                {
+                    var message = "Migration Validation Error: Failed to parse the KVS key - " + key;
+
+                    ActorTrace.Source.WriteErrorWithId(
+                        this.TraceType,
+                        this.traceId,
+                        message);
+
+                    this.servicePartition.ReportPartitionHealth(new HealthInformation(this.TraceType, message, HealthState.Error));
```

Comment:
I am thinking we should not be reporting partition health here as there could be other keys(say _RejectWrites_) which we are not interested. Tracing Error and moving forward is fine. Agree?

Reply:
Agreed.